dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/hdfs/ScanContext.java

```diff
@@ -22,13 +22,23 @@
 import java.io.Writer;
 import java.time.Duration;
 import java.time.Instant;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
 import org.apache.commons.csv.CSVPrinter;
 import org.apache.hadoop.fs.FileStatus;
 import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.hdfs.DFSClient;
+import org.apache.hadoop.hdfs.DistributedFileSystem;
+import org.apache.hadoop.hdfs.protocol.HdfsConstants.StoragePolicy;
+import org.apache.hadoop.hdfs.protocol.HdfsFileStatus;
 
 final class ScanContext implements Closeable {
```

Comment:
The attributes `numFilesByListStatus` and `timeSpentInListStatus` are guarded by `LOCK1` inside `listDirectory`.

Should they be guarded similarly in `getFormattedStats`?

Reply:
Despite the fact that thus is a diagnostic/log method, you're absolutely right.